patroni/ha.py

```diff
@@ -270,6 +272,23 @@ def demote(self, delete_leader=True):
 
     def process_manual_failover_from_leader(self):
         failover = self.cluster.failover
+
+        if failover.planned_at:
+            now = utcnow_timezone_aware()
+            delta = (failover.planned_at - now).total_seconds()
+
+            if delta > 10:
+                logging.debug('Awaiting failover, {0}, {1}, {2}'.format(now.isoformat(),
+                              failover.planned_at.isoformat(),
+                              delta))
+                return
+            elif delta < -15:
+                logger.info('Found a stale failover value, cleaning up: {}'.format(failover.planned_at))
+                self.dcs.manual_failover('', '', self.cluster.failover.index)
+                return
+            time.sleep(max(delta, 0))
```

Comment:
What if a failover is scheduled 24h in the future and the process doing this sleep() dies for some reason: what will happen to it upon restart?  Will it notice a failover pending again and sleep for as long as it still should?